manifests/server/config.pp

```diff
@@ -155,11 +155,25 @@
 
   # autosign file
   if $::puppet::server_ca and ! is_bool($puppet::autosign) {
-    file { $puppet::autosign:
-      ensure => file,
-      owner  => $puppet::user,
-      group  => $puppet::group,
-      mode   => $puppet::autosign_mode,
+    if $::puppet::autosign_content and !empty($puppet::autosign_entries){
```

Comment:
I think the following should work (but untested). The benefit is that you have one place to define the autosign file resource.

```puppet
if $::puppet::server::autosign_content {
  if !empty($::puppet::server::autosign_entries) {
    fail('Cannot set both autosign_content and autosign_entries')
  }
  $autosign_content = $::puppet::server::autosign_content
} elsif !empty($::puppet::server::autosign_entries) {
  $autosign_content = template('puppet/server/autosign.conf.erb')
} else {
  $autosign_content = undef
}

file { $::puppet::server::autosign:
  ensure  => file,
  owner   => $::puppet::server::user,
  group   => $::puppet::server::group,
  mode    => $::puppet::server::autosign_mode,
  content => $autosign_content,
}
```